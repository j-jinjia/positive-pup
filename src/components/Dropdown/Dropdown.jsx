import "./Dropdown.scss";
import { useState } from "react";
import React from "react";

const Dropdown = ({ question, answer }) => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleDropdown = () => {
    setIsOpen(!isOpen);
  };

  const mainClass = isOpen
    ? "dropdown__main dropdown__main--active"
    : "dropdown__main";

  const answerArr = answer.split("\n");
  const answerText = answerArr.map((answer, i) => (
    <React.Fragment key={answer}>
      <p className="dropdown__main-text">{answer}</p>
      {i !== answerArr.length - 1 && <br />}
    </React.Fragment>
  ));

  return (
    <div className="dropdown">
      <label className="dropdown__heading-label" htmlFor="dropdown">
        {"Click to view answer to " + question}
      </label>
      <button
        id="dropdown"
        className="dropdown__heading"
        onClick={toggleDropdown}
      >
        <h3 className="dropdown__heading-title">{question}</h3>
        <p className="dropdown__heading-action">{isOpen ? "-" : "+"}</p>
      </button>
      <div className={mainClass}>{answerText}</div>
    </div>
  );
};

export default Dropdown;
