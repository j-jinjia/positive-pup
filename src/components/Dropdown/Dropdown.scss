@use "../../styles/variables" as *;

.dropdown {
  width: 100%;
  max-width: 62rem;
  margin-bottom: 0.75rem;
  background-color: transparent;
  text-align: left;

  &__heading {
    width: 100%;
    border: none;
    background-color: $colour-septenary;
    padding: 1rem 2rem;
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 0.5rem;
    &-title,
    &-action {
      font-size: inherit;
      font-weight: normal;
    }
    &:hover {
      cursor: pointer;
    }
    &-label {
      display: none;
    }
  }

  &__main {
    display: none;
    padding: 1rem 2rem;
    padding-bottom: 0;
    border: 1px solid $colour-septenary;
    &-text {
      color: $colour-quinary;
    }
    &--active {
      display: block;
    }
  }
}

@media screen and (min-width: $tablet-breakpoint) {
  .dropdown {
    &__heading {
      padding: 1rem 4rem;
      font-size: $font-size-smaller;
    }

    &__main {
      padding: 1rem 4rem;
    }
  }
}
