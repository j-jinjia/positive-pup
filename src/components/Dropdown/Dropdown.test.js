import { customRender } from "../../utils/testUtils";
import Dropdown from "./Dropdown";
import faqData from "../../assets/mockData/faqsData";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

it("should match Dropdown component snapshot", () => {
  const dropdownData = faqData[0];
  const { container } = customRender(
    <Dropdown question={dropdownData.question} answer={dropdownData.answer} />
  );

  expect(container).toMatchSnapshot();
});

it("if the dropdown is clicked, the text should become visible", () => {
  customRender(<Dropdown question={"question"} answer={"answer"} />);
  const showToggle = screen.getByRole("button");
  userEvent.click(showToggle);
  const answer = screen.getByText("answer");
  expect(answer).toBeInTheDocument();
});
