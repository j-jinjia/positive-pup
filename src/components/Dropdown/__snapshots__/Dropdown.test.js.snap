// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Dropdown component snapshot 1`] = `
<div>
  <div
    class="dropdown"
  >
    <label
      class="dropdown__heading-label"
      for="dropdown"
    >
      Click to view answer to Why should I choose The Positive Pup?
    </label>
    <button
      class="dropdown__heading"
      id="dropdown"
    >
      <h3
        class="dropdown__heading-title"
      >
        Why should I choose The Positive Pup?
      </h3>
      <p
        class="dropdown__heading-action"
      >
        +
      </p>
    </button>
    <div
      class="dropdown__main"
    >
      <p
        class="dropdown__main-text"
      >
        That is a great question! Finding the right trainer isnâ€™t easy and can be a scary prospect.  The last thing you want to do is waste money on the wrong solution and cause any long term problems for your dog, right?
      </p>
      <br />
      <p
        class="dropdown__main-text"
      >
        Because of this it will be reassuring to you to know that during my time at the Victoria Stilwell Academy, I have had the benefit of training from some of the best people within the industry. Even more encouraging is that the positive reinforcement method I have been trained in is backed heaps of by scientific evidence and is proven to be the best method available to achieve outstanding results.
      </p>
      <br />
      <p
        class="dropdown__main-text"
      >
        To add to this you can guarantee that your dog will enjoy these sessions and will love to learn!
      </p>
    </div>
  </div>
</div>
`;
